test/test_torch.py

```diff
@@ -11226,11 +11249,6 @@ def test_vander(self, device):
         with self.assertRaisesRegex(RuntimeError, "x must be a one-dimensional tensor."):
             torch.vander(torch.stack((x, x)))
 
-        # This passes on the xla backend
-        if device != 'xla':
```

Comment:
what is this change for?

Reply:
Similar to `rot90`,`vander` also uses `flip` internally.

https://github.com/pytorch/pytorch/blob/25e6129c5258eb966c8e843ab43e7b4ebaa1b8de/aten/src/ATen/native/TensorFactories.cpp#L920-L946

The removed lines, used to assert that passing a complex dtype will raise a runtime error on `CPU` and `CUDA` backends (not sure why it didn't on `xla`)

However with this PR that won't be the case and thus have removed it.

Reply:
If you're going to make Vander work for complex values you need to add a test for Vander's behavior on complex values, too. You could add a case, for example, to test_vander_types (below), which compares with NumPy as I'm recommending the two tests you're adding do, too. Your case would look something like:

```
if dtype is complex:
  x = [[1 + 1j, 2 - 2j...], ...]]
```

Reply:
`vander` uses `at::cumprod` which doesn't support complex dtypes on `cuda` .

https://github.com/pytorch/pytorch/blob/ff9a809ccd5bd61f18dfaa04d41b681e600b3a02/aten/src/ATen/native/TensorFactories.cpp#L939-L942

So have added test for only CPU.

Reply:
That's good but then the check should be `if self.device_type == `'cuda'` and you should add a comment clarifying why the CUDA device type raises.